### PR TITLE
Fix segv: use root_id_ when plugin_ is nullptr.

### PR DESCRIPTION
**Pull request by jplevyak**
Signed-off-by: John Plevyak <jplevyak@gmail.com>

Fixes: 

https://github.com/envoyproxy/envoy-wasm/issues/327

### CODE REVIEW

**Comment by jplevyak** [comment]
Also fixes: https://github.com/envoyproxy/envoy-wasm/issues/326

**Comment by mandarjog** [comment]
@jplevyak none of our proxy tests caught this problem. Can we add tests that will bring out this issue? @kyessenov  ?

**Comment by jplevyak** [comment]
Test added.  PTAL

File: test/extensions/wasm/test_data/logging_cpp.cc [hunk 1]
```diff
@@ -20,7 +20,10 @@ extern "C" EMSCRIPTEN_KEEPALIVE uint32_t proxy_on_start(uint32_t, uint32_t) {
 }
 
 extern "C" EMSCRIPTEN_KEEPALIVE void proxy_on_tick(uint32_t) {
-  logInfo("test tick logging");
+  const char* root_id = nullptr;
+  size_t size;
+  proxy_get_property("plugin_root_id", sizeof("plugin_root_id") - 1, &root_id, &size);
+  logInfo("test tick logging" + std::string(root_id, size));
```
Comment (PiotrSikora) [inline test/extensions/wasm/test_data/logging_cpp.cc]: `test/extensions/wasm/wasm_test.cc` expects `"test tick logging"`:
```
EXPECT_CALL(*context, scriptLog_(spdlog::level::info, Eq("test tick logging")));
```
so how is this passing? Is `root_id` empty?

Comment (jplevyak) [reply test/extensions/wasm/test_data/logging_cpp.cc]: root_id is empty.   The bug was that just asking for the root_id was causing a SEGV.  That is now fixed.

**Pull request merged**